Added particle output to file; added Waterbag comments

cemitch99 opened the pull request:
Added output of initial particle data to the file "particles.txt" inside "initDist" (for debugging purposes), and added some lines of documentation to the function "Waterbag" for initial bunch generation.

cemitch99 committed:
Added particle output to file for debugging purposes, and added
comments for documentation of "Waterbag" beam generation.

ax3l commented on src/particles/distribution/Waterbag.H:
```diff
@@ -66,6 +66,8 @@ namespace distribution
             constexpr amrex::ParticleReal pi = 3.14159265358979_prt;
+
+//    Generate six standard normal random variables using Box-Muller:
             u1 = amrex::Random(engine);
```
```suggestion
            // Generate six standard normal random variables using Box-Muller:
            u1 = amrex::Random(engine);
```

ax3l commented on src/particles/distribution/Waterbag.H:
```diff
@@ -81,13 +83,17 @@ namespace distribution
             g5 = ln1*cos(2_prt*pi*u2);
             g6 = ln1*sin(2_prt*pi*u2);
+
+//     Normalize to produce uniform samples on the unit sphere:
             norm = sqrt(g1*g1+g2*g2+g3*g3+g4*g4+g5*g5+g6*g6);
```
```suggestion
            // Normalize to produce uniform samples on the unit sphere:
            norm = sqrt(g1*g1+g2*g2+g3*g3+g4*g4+g5*g5+g6*g6);
```

ax3l commented on src/particles/distribution/Waterbag.H:
```diff
@@ -81,13 +83,17 @@ namespace distribution
             g6 /= norm;
+
+//     Scale to produce uniform samples in a ball (unit variance):
             u1 = amrex::Random(engine);
```
```suggestion
            // Scale to produce uniform samples in a ball (unit variance):
            u1 = amrex::Random(engine);
```

ax3l commented on src/particles/distribution/Waterbag.H:
```diff
@@ -97,6 +103,7 @@ namespace distribution
             pt = g6*u2;
 
+//     Transform to produce the desired second moments/correlations:
             root = sqrt(1.0_prt-m_muxpx*m_muxpx);
```
```suggestion
            // Transform to produce the desired second moments/correlations:
            root = sqrt(1.0_prt-m_muxpx*m_muxpx);
```

ax3l reviewed (commented):
Just some formatting

ax3l committed:
Align comments

ax3l commented on src/ImpactX.cpp:
```diff
@@ -286,6 +286,9 @@ namespace impactx
                   px.push_back(ipx);
                   py.push_back(ipy);
                   pt.push_back(ipt);
+                  amrex::PrintToFile("particles.txt") << ix << " " << iy << " ";
+                  amrex::PrintToFile("particles.txt") << it << " " << ipx << " ";
+                  amrex::PrintToFile("particles.txt") << ipy << " " << ipt << " " << std::endl;
```
Since we plot this at init, shall we call this `initial_beam.txt` for now?
```suggestion
                  amrex::PrintToFile("initial_beam.txt") << ix << " " << iy << " ";
                  amrex::PrintToFile("initial_beam.txt") << it << " " << ipx << " ";
                  amrex::PrintToFile("initial_beam.txt") << ipy << " " << ipt << " " << std::endl;
```

cemitch99 committed:
Update src/ImpactX.cpp

Co-authored-by: Axel Huebl <axel.huebl@plasma.ninja>

ax3l committed:
Add Missing Include

Adding this makes builds more robust in case AMReX restructures
its internal cross-includes in the future.

ax3l reviewed (approved):
Thank you! :)
